### PR TITLE
Enforce no forks

### PR DESCRIPTION
**Pull request by frozenfrank**
## Overview
Leverages refactors from #497 to provide detection of "forked" repos required by https://github.com/softwareconstruction240/softwareconstruction/issues/138.

## Testing

- [x] Added/updated unit tests
- [ ] Tested edge cases
- [ ] Manual testing (if needed)

## Additional Notes

This is the first step in making it significantly easier for students to clone the starter code, while also blocking easy attempts at cheating.

### CODE REVIEW

File: src/test/java/edu/byu/cs/util/RepoUrlValidatorTest.java [hunk 1]
```diff
@@ -137,4 +140,60 @@ void adminSubmissionsAreNotCleaned() throws GradingException, IOException {
 //        Assertions.assertEquals(originalUrl, grader.gradingContext.repoUrl());
     }
 
+    //  TODO: Convert these to @ParameterizedTest
```
Comment (frozenfrank) [inline src/test/java/edu/byu/cs/util/RepoUrlValidatorTest.java:143]: @19mdavenport Can you help convert these to `@ParameterizedTests`? I was able to do the code part of it, but I ran into compile issues. It seems the dependencies required to use the parameterized portions are not available in the project, even though I copied this directly from another place where it is used:

https://github.com/softwareconstruction240/autograder/blob/9ea6e4d3ed4b3b9f86b59c939e5cea2897ecdc30/src/main/resources/phases/phase1/passoff/chess/game/MakeMoveTests.java#L7

Comment (19mdavenport) [reply src/test/java/edu/byu/cs/util/RepoUrlValidatorTest.java:143]: Is this required for this PR to be merged, or could this be a separate issue? I'm busy with the schedule and other things today but I would be happy to help with this later

**Review by 19mdavenport — APPROVED** [review]
Looks good so far

File: src/test/java/edu/byu/cs/util/RepoUrlValidatorTest.java (hunk 1 above)
Comment (frozenfrank) [reply src/test/java/edu/byu/cs/util/RepoUrlValidatorTest.java:143]: This is not required. I'll turn it into a separate issue.

Comment (frozenfrank) [reply src/test/java/edu/byu/cs/util/RepoUrlValidatorTest.java:143]: @19mdavenport  Done. See https://github.com/softwareconstruction240/autograder/issues/504
